packages/ui-buttons/src/CloseButton/index.tsx

```diff
@@ -113,10 +113,13 @@ class CloseButton extends Component<CloseButtonProps> {
     } = this.props
 
     return (
-      <span {...passthroughProps(props)} css={styles?.closeButton}>
+      <span
+        {...passthroughProps(props)}
+        css={styles?.closeButton}
+        ref={this.handleRef}
+      >
         <BaseButton
           renderIcon={IconXSolid}
-          elementRef={this.handleRef}
           interaction={this.interaction}
```

Comment:
This is the general bug pattern that was uncovered by switching from React's deprecated to our version of `findDOMNode`: `ref` was not pointing to the top DOM element in the component

Reply:
Just on first glance: Looking at this makes me worried that fixes like this (moving the whole `handleRef` up) might cause breaking changes :( When we added the `ref`-s, most components already had their `elementRef` in place. Even for CloseButton, if you check older versions (e.g. in V7), you can see that the elementRef was always giving back the BaseButton, not the wrapper. If someone relies on this in their tests for example, than it is a breaking change for them. So the `elementRef`s should stay where they were imho.

Reply:
hmmm then some of the tests will fail and we might not be able to fix them. An example: If we leave the `elementRef` of `ToggleButton` as is (see this changeset) then the `should display a tooltip without hover/focus when isShowingTooltip is true` will fail. This is because we find the tooltip like this:

```
const tooltip = await button.findTooltipContent()
```
this code does calls (after some loops) `locator(ToggleButton.selector,...).find()`. `ToggleButton.selector` is added to the component by `testable.ts`, that uses our `findDOMNode` to find out where to add it. 
And since our `findDOMNode` return the value of our `ref` prop, it wont find the tooltip, since its one level up in the DOM.

TL/DR: locators search in the DOM subtree returned by our `findDOMNode` so if its not top level it wont see whats above.

So we can fix it the following ways:
1. We move the refs up as in this commit. We can treat the previous solution as a bug since it worked like in this commit before we added the `ref`s (the values I added here are the same that `ReactDOM.findDOMNode` would return). 
2. We introduce some new prop to these components that points to the top of them and somehow access it after its mounted.
3. We rewrite the tests to get the parent of the node, e.g. call `button.getParentNode()`

What do you think? I'd prefer 1), maybe 2)

Reply:
As we discussed it I'll put back `elementRef`s to their original place and just move `ref`s